packages/rescript-mui-material/src/types/CommonProps.res

```diff
@@ -98,19 +98,139 @@ type ariaProps = {
   ariaRowspan?: int,
   @as("aria-setsize")
   ariaSetsize?: int,
+  /**
+    * The ARIA role attribute of the element.
+    */
+  role?: string,
 }
 
-type t_NoId = {
-  // Cannot spread whole dom props since overwriting is not possible.
-  ...ariaProps,
+type autoCapitalize =
+  | @as("none") None
+  | @as("sentences") Sentences
+  | @as("words") Words
+  | @as("characters") Characters
+
+type autoCorrect = | @as("on") On | @as("off") Off
+type dir = | @as("ltr") Ltr | @as("rtl") Rtl | @as("auto") Auto
+type contentEditable = | @as(true) True | @as(false) False | @as("inherit") Inherit
+
+/**
+https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/enterkeyhint
+ */
+type enterKeyHint =
+  | @as("enter") Enter
+  | @as("done") Done
+  | @as("go") Go
+  | @as("next") Next
+  | @as("previous") Previous
+  | @as("search") Search
+  | @as("send") Send
+
+/**
+https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode
+ */
+type inputMode =
+  | @as("none") None
+  | @as("text") Text
+  | @as("decimal") Decimal
+  | @as("numeric") Numeric
+  | @as("tel") Tel
+  | @as("search") Search
+  | @as("email") Email
+  | @as("url") Url
+
+type globalAttributes = {
+  autoCorrect?: autoCorrect,
+  /**
+    * If `true`, the element is focused during the first mount.
+    */
+  autoFocus?: bool,
   className?: string,
-  style?: ReactDOM.Style.t,
-  ref?: ReactDOM.domRef,
+  contentEditable?: contentEditable,
+  contextMenu?: string,
   @as("data-testid") dataTestId?: string,
+  dir?: dir,
+  draggable?: bool,
+  enterKeyHint?: enterKeyHint,
+  hidden?: bool,
+  inert?: bool,
+  lang?: string,
+  nonce?: string,
+  slot?: string,
+  spellCheck?: bool,
   /**
-    * @default 0
-    */
+   * @default 0
+   */
   tabIndex?: int,
+  // title is defined & overwritten elsewhere
+  style?: ReactDOM.Style.t,
+}
+
+/**
+ https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#wrap
+ */
+type wrap =
+  | @as("hard") Hard
+  | @as("soft") Soft
+  | @as("off") Off
+
+type inputTextareaProps = {
+  /**
+    * This prop helps users to fill forms faster, especially on mobile devices.
+    * The name can be confusing, as it's more like an autofill.
+    * You can learn more about it [following the specification](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill).
+    */
+  autoComplete?: string,
+  autoCapitalize?: autoCapitalize,
+  // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-inputmode
+  inputMode?: inputMode,
+  /**
+    * Type of the `input` element. It should be [a valid HTML5 input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).
+    */
+  @as("type")
+  type_?: string,
+  /**
+    * The maximum value the user can enter in the input. See [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/max) for more details.
+    */
+  max?: string,
```

Comment:
@fhammerschmidt 

The `min` and `max` props technically are not just `int`, maximum/minimum values can be other values too: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/max

Additionally the `step` is probably better off as a `float`: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/step#syntax

Since this is a breaking change, should I leave it out? Otherwise I can make it an unboxed type that is `string | float`

Reply:
Yeah it mainly depends on the type and thus is probably a good fit for unboxed.
Let's merge this first.